providers/common/digests/sha.c

```diff
@@ -0,0 +1,63 @@
+/*
```

Comment:
Is it worthwhile naming this file `sha2.c`?
Apart from the names, there isn't a lot of similarity between SHA1, SHA2 and SHA3.

Reply:
Done